### PR TITLE
Validate user email and password on creation

### PR DESCRIPTION
**Pull request by dimroc**

### CODE REVIEW

File: store/models/user.go [hunk 1]
```diff
@@ -15,10 +16,21 @@ type User struct {
...
-		return User{}, errors.New("Must enter an email or password")
+	if len(email) == 0 {
+		return User{}, errors.New("Must enter an email")
+	}
+
+	if !emailRegexp.MatchString(email) {
+		return User{}, errors.New("Invalid email format")
+	}
+
+	if len(plainPwd) < 8 || len(plainPwd) > 1028 {
```
Comment (rupurt) [inline store/models/user.go]: How do you feel about making these values config variables?

Comment (se3000) [reply store/models/user.go]: Let's add this as a follow up story.

File: store/models/user.go [hunk 2]
```diff
@@ -15,10 +16,21 @@ type User struct {
 	CreatedAt      Time   `json:"createdAt" storm:"index"`
 }
 
+// https://davidcel.is/posts/stop-validating-email-addresses-with-regex/
```
Comment (se3000) [inline store/models/user.go]: > But guess what? You can use pretty much any character you want if you escape it by surrounding it in quotes. For example, "Look at all these spaces!"@example.com is a valid email address.

🤯

Still happy to stick with regex for now.

File: store/models/user.go (hunk 1 above)
Comment (se3000) [reply store/models/user.go]: https://www.pivotaltracker.com/story/show/159445522

**Pull request merged**